core/capabilities/ccip/configs/solana/chain_writer.go

```diff
@@ -0,0 +1,50 @@
+package solana
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/chainwriter"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
+)
+
+//go:embed ccip_router.json
+var ccipRouter string
+
+func GetSolanaChainWriterConfig(fromAddress string) (chainwriter.ChainWriterConfig, error) {
+	// TODO once on-chain account lookup address are available, the config will be updated
+
+	// check fromAddress
+	_, err := solana.PublicKeyFromBase58(fromAddress)
+	if err != nil {
+		return chainwriter.ChainWriterConfig{}, fmt.Errorf("invalid from address %s: %w", fromAddress, err)
+	}
+
+	// validate CCIP Router IDL, errors not expected
+	var idl codec.IDL
+	if err = json.Unmarshal([]byte(ccipRouter), &idl); err != nil {
+		return chainwriter.ChainWriterConfig{}, fmt.Errorf("unexpected error: invalid CCIP Router IDL, error: %w", err)
+	}
+
+	solConfig := chainwriter.ChainWriterConfig{
```

Comment:
When the rest of the address like `routerProgram ` and `commonAddressesLookupTable` etc. become available, will they be hard coded here, or provided by tooling team ?

Reply:
I think should be provided by some sort of tooling since addresses will be different for testnets than mainnet